zerver/tornado/views.py

```diff
@@ -34,8 +34,9 @@ async def wrapped(*args: P.args, **kwargs: P.kwargs) -> T:
 
 
 @internal_notify_view(True)
-def notify(request: HttpRequest) -> HttpResponse:
-    in_tornado_thread(process_notification)(orjson.loads(request.POST["data"]))
+@has_request_variables
+def notify(request: HttpRequest, data: str = REQ()) -> HttpResponse:
+    in_tornado_thread(process_notification)(orjson.loads(data))
```

Comment:
I think we want `REQ` to take care of any JSON loading process here -- something like, `validator=check_dict` will do that. Though it might be even better to actually check the type, that'd be only really useful if we improved the types on `check_notification`. @PIG208 FYI.

Reply:
Yes, that makes sense. I'm guessing that `check_notification` in your note was meant to be `process_notification`, as I can't find the former through `git-grep` in the codebase?

And looks like, without improving the types on `process_notification`, the type annotation here would currently be ...

```python
data: Mapping[str, Any] = REQ(json_validator=check_dict([])
```

Reply:
Yep, that was a typo.

Reply:
I've updated for the above type annotation, which removes the `orjson` loading / processing.